### PR TITLE
feat: annotate each path with other release versions

### PR DESCRIPTION
**Pull request by cmars**
API documentation needs to represent a "changelog" per resource
endpoint: a list of versions in which a change to a given resource
endpoint was released.

This change adds a list of other resource release versions in which a
particular path was declared for such navigation purposes.

Implements #78.

### CODE REVIEW

File: resource.go [hunk 1]
```diff
@@ -139,31 +144,49 @@ func LoadResourceVersions(epPath string) (*ResourceVersions, error) {
...
+	for _, releases := range pathReleases {
+		sort.Sort(releases)
+	}
+	// Sort the resources themselves by version
+	sort.Sort(resourceVersionSlice(resourceVersions.versions))
+	// Annotate each path in each resource version with the other change
+	// versions affecting the path. This supports navigation across versions.
+	for _, rc := range resourceVersions.versions {
+		for path, pathInfo := range rc.Paths {
+			pathInfo.ExtensionProps.Extensions[ExtSnykApiReleases] = pathReleases[path].Strings()
```
Comment (jcsackett) [inline resource.go:186]: Is there a bit of code that's automatically adding extension sections to the spec output? I see the data added here, and I see a string representation utility function at the end of the PR, but no updates to have this put in spec, so guessing that's automatic?

File: resource.go [hunk 2]
```diff
@@ -139,31 +144,49 @@ func LoadResourceVersions(epPath string) (*ResourceVersions, error) {
 }
 
 func LoadResourceVersionsFileset(specYamls []string) (*ResourceVersions, error) {
-	var eps ResourceVersions
+	var resourceVersions ResourceVersions
```
Comment (jcsackett) [inline resource.go:147]: Good renaming; what was `eps` for?

**Review by jcsackett — COMMENTED** [review]
LGTM! Questions in the diff for my own edification (unless it turns out I stumbled across something missing 🙂)

File: resource.go (hunk 2 above)
Comment (cmars) [reply resource.go:147]: Endpoints.

File: resource.go (hunk 1 above)
Comment (cmars) [reply resource.go:186]: These extensions are basically being added on load, which gets merged into the compiled result by `vervet.Merge`.

The utility function is used to read the value of extensions provided in the source OpenAPI spec.

**Review by cmars — COMMENTED** [review]
Replies to comments.

**Pull request closed without merging**